File: FrontEnd/lib/domain/interfaces/usecases/user/i_update_user_preferences_use_case.dart
/// @file i_update_user_preferences_use_case.dart
/// @description Interfaz: Actualiza las preferencias del usuario.
/// @module User
/// @layer Domain
library;

import '../../../../domain/dtos/user/update_preferences_request_dto.dart';
import '../../../../domain/entities/user.dart';

abstract interface class IUpdateUserPreferencesUseCase {
  /// Actualiza las preferencias del usuario autenticado.
  ///
  /// [param] dto — preferencias a actualizar.
  /// [returns] [User] con las preferencias actualizadas.
  Future<User> execute(UpdatePreferencesRequestDto dto);
}

File: BackEnd/src/domain/interfaces/usecases/user/IUpdateUserPreferencesUseCase.ts
/**
 * @file IUpdateUserPreferencesUseCase.ts
 * @description Interfaz del caso de uso para actualizar las preferencias del usuario.
 * @module User
 * @layer Domain
 */

import type { UpdatePreferencesRequestDto } from '../../../dtos/user/update-preferences-request.dto.js';
import type { UserResponseDTO } from '../../../dtos/user/user-response.dto.js';

export interface IUpdateUserPreferencesUseCase {
  /**
   * @param userId — ID del usuario autenticado.
   * @param dto — Preferencias a actualizar (parcial).
   * @returns Datos públicos del usuario con preferencias actualizadas.
   * @throws {NotFoundException} Si el usuario no existe.
   */
  execute(userId: string, dto: UpdatePreferencesRequestDto): Promise<UserResponseDTO>;
}
